src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversion.cs

```diff
@@ -357,8 +357,10 @@ internal MethodSymbol Method
                         return analysis.Operator;
                     }
 
-                    if (uncommonData is DeconstructionUncommonData deconstruction
-                        && deconstruction.DeconstructMethodInfo.Invocation is BoundCall call)
+                    if (uncommonData is DeconstructionUncommonData
+                    {
+                        DeconstructMethodInfo: { Invocation: BoundCall call }
+                    } deconstruction)
```

Comment:
strongly worry about expanding simple binary expression into what feels like more complex patterns automatically.

Reply:
The expansion is not automatic since this is a refactoring.